### PR TITLE
Added possibility of installing Hawkular APM

### PR DESCRIPTION
**Pull request by jpkrohling**

### CODE REVIEW

File: ansible/helloworld-msa.yml [hunk 1]
```diff
@@ -133,13 +135,26 @@
...
       failed_when: "'exists' not in command_result.stderr and command_result.rc != 0"
       changed_when: "'exists' not in command_result.stderr"
       tags:
         - openshift
+        - opentracing
         - zipkin
 
-#Install Jenkins
+    - name: Install Hawkular APM
+      shell: "oc create -f /home/jpkroehling/Projects/jboss-dockerfiles-hawkular/openshift-templates/hawkular-apm-server-deployment.yml"
```
Comment (jpkrohling) [inline ansible/helloworld-msa.yml]: This needs to change!

**Comment by jpkrohling** [comment]
Do not merge yet: this is just to start a discussion.

**Comment by jpkrohling** [comment]
@rafabene , @objectiser , I did some changes to the Ansible playbook, to install a Hawkular APM server instead of ZipKin by default. Besides this change, a small change in the value of an env var is required, as seen on the referenced PR for Aloha. 

If this is something that would be acceptable to the Hello World MSA, I'd also change the instructions, swapping the positions of ZipKin and Hawkular APM. Note that the *only* change required on the modules is the ZipKin server location. 

On future PRs, I'll change the modules (Aloha, Bonjour) to use OpenTracing instead of ZipKin and I'd like to keep that as a separate PR.

**Comment by objectiser** [comment]
@jpkrohling LGTM :+1:  Although I haven't used ansible before so haven't tried it out. If you can point me to instructions I'll try it out.
Don't worry - assume these are the instructions: https://cdn.rawgit.com/redhat-helloworld-msa/helloworld-msa/master/readme.html#__option_1_install_all_microservices_using_ansible - will try it out later today.

**Comment by jpkrohling** [comment]
Sure, we'll either need https://github.com/jboss-dockerfiles/hawkular-apm/pull/24 merged, or you'll need to download that file and change https://github.com/redhat-helloworld-msa/helloworld-msa/pull/21/files#diff-67ddaf9aa26159a82e5173928fb60e45R147 to point to your local version of this file. Then:

```
cd ansible
docker-machine create -d virtualbox --virtualbox-memory 8192 --virtualbox-cpu-count 4 --engine-insecure-registry 172.30.0.0/16 openshift
oc cluster up --docker-machine=openshift
```

Download this https://paste.fedoraproject.org/488396/81025614/raw/ as `vars.yml`, changing the OpenShift IP to the IP that the `oc cluster up` shows you, as well as the `domain` property. After that, just run `ansible-playbook helloworld-msa.yml`.

This is basically the "Option 2" of the "Prepare Environment" + "Option 1" of the "install microservices" steps from the MSA Hello World readme.

**Comment by jpkrohling** [comment]
@rafabene , this is the first step in having Hawkular APM on Hello World MSA. Updates to the individual components will follow, once this PR is merged.

This last state differs from the previous iterations, as no "extra" services are now installed by default. As such, APM is not replacing ZipKin.

**Comment by rafabene** [comment]
Thanks. Merged!

**Pull request merged**